setup.sh

```diff
@@ -84,601 +120,224 @@ vercomp () {
+      return 0
     fi
     if ((10#${ver1[i]} < 10#${ver2[i]}))
     then
-      return 2
+      return 1
     fi
   done
   return 0
 }
 
-findOSType()
-{
-  osType=$(uname)
-  if [ "$osType" == "Linux" ]; then
-    if [ -f /etc/os-release ]; then
-      distro=$(awk -F= '/^NAME/{print $2}' /etc/os-release)
-    else
-      distro=$(ls -d /etc/[A-Za-z]*[_-][rv]e[lr]* | grep -v "lsb" | cut -d'/' -f3 | cut -d'-' -f1 | cut -d'_' -f1)
-    fi
-    CURRENT_OS=$(echo $distro | tr 'a-z' 'A-Z' | cut -d' ' -f1 | tr -d '"')
-  fi
-}
-
-isOSSupported()
-{
-  case "$CURRENT_OS" in
-    "UBUNTU" | "DEBIAN")
-    {
-      OS_IS_SUPPORTED=1
-    } ;;
-    *)
-    {
-      echo
-      writeBold "$YELLOW" "[✖] Your OS ($CURRENT_OS) is not officially supported."
-      write               "    This means we didn't thoroughly test Kuzzle on your"
-      write               "    system. It is likely to work, so you may want to continue."
-      OS_IS_SUPPORTED=0
-      while [[ "$proceedNotSupported" != [yYnN] ]]
-      do
-        writeBold "[❓] Do you want to continue? (y/N)"
-        echo -n "> "
-        read proceedNotSupported trash
-        case "$proceedNotSupported" in
-          [yY])
-            echo
-            writeBold "$GREEN" "Great! Join us on Gitter if you need help (https://gitter.im/kuzzleio/kuzzle)"
-            ;;
-          [nN] | '')
-            if [ -n $DL_BIN ]; then
-              echo
-              writeBold "[❓] Would you like to be notified when your OS is supported? (y/N)"
-              write      "   This will increase the chances that your OS will be supported in the future."
-              echo -n "> "
-              read notifyTeam trash
-              case "$notifyTeam" in
-                [yY])
-                  writeBold "[❓] Ok. What is your email address?"
-                  echo -n "> "
-                  read email trash
-                  $DL_BIN $UL_OPTS '{"type": "notify-when-os-supported", "uid": "'$UUID'", "os": "'$CURRENT_OS'", "email": "'$email'"}' $ANALYTICS_URL &> /dev/null
-                  ;;
-                *)
-                  echo
-                  writeBold "$BLUE" "Ok. We encourage you to get in touch with the team ($SUPPORT_MAIL)"
-                  writeBold "$BLUE" "to request support for your system."
-                  echo
-                  ;;
-              esac
-              exit 2
-            fi
-            ;;
-          *)
-            echo
-            writeBold "$RED" "[✖] Please, answer Y or N."
-            ;;
-        esac
-      done
-    } ;;
-  esac
-}
+prerequisite() {
+  local ERROR=0
 
-exitIfNotRoot()
-{
-  if [ "$EUID" -ne 0 ]; then
-    echo
-    writeBold "$YELLOW" "[✖] This script needs to be executed with root privileges to be able to perform an install on your system"
-    write               "    Try typing: sudo $0"
-    echo
-    exit 1
+  # Check internet connection
+  if [ $(eval "$KUZZLE_CHECK_DOCKER_COMPOSE_YML_HTTP_STATUS_CODE") -ne 200 ]; then
```

Comment:
If we get a http status code > 100, then it means that we DO have an internet connection.
As it is, the script will complain that there is no internet connection if the docker-compose file is unavailable for whatever reason. This may because of a 404. Or a 503. That error is misleading and can be cause of complaints from users.

That test should instead check the return values of either curl or wget instead of the http code.
For instance, curl exits with code: `89     No connection available, the session will be queued` 
While wget exits with `4   Network failure.`